packages/core/src/components/fields/ObjectField.js

```diff
@@ -83,7 +83,7 @@ class ObjectField extends Component {
 
   onPropertyChange = (name, addedByAdditionalProperties = false) => {
     return (value, errorSchema) => {
-      if (!value && addedByAdditionalProperties) {
+      if (typeof value === "undefined" && addedByAdditionalProperties) {
```

Comment:
Let's do this, because the variable `value` is always going to be defined (whereas `value`'s value may be equal to undefined)

```suggestion
      if (value === undefined && addedByAdditionalProperties) {
```